voicevox_engine/tts_pipeline/tts_engine.py

```diff
@@ -590,6 +591,41 @@ def synthesize_wave(
         wave = raw_wave_to_output_wave(query, raw_wave, sr_raw_wave)
         return wave
 
+    def synthesize_wave_stream(
+        self,
+        query: AudioQuery,
+        style_id: StyleId,
+        enable_interrogative_upspeak: bool = True,
+    ) -> tuple[int, Generator[NDArray[np.float32], None, None]]:
+        """生成音声全体のフレーム数と音声波形を生成する同期ストリームを返す"""
+        valid_chunk_size = 94  # 1sec * 24000Hz / 256frame
+        query = copy.deepcopy(query)
+        query.accent_phrases = _apply_interrogative_upspeak(
+            query.accent_phrases, enable_interrogative_upspeak
+        )
+
+        phoneme, f0 = _query_to_decoder_feature(query)
+        audio_feature = self._core.safe_generate_full_intermediate(
+            phoneme, f0, style_id
+        )
+
+        def wave_generator() -> Generator[NDArray[np.float32], None, None]:
+            for render_start in range(0, len(audio_feature), valid_chunk_size):
+                render_end = min(render_start + valid_chunk_size, len(audio_feature))
+                slice_start = render_start
+                slice_end = render_end + 2 * self._core.margin_width
+                feature_segment = audio_feature[slice_start:slice_end, :]
+                raw_wave_with_margin, sr_raw_wave = (
+                    self._core.safe_render_audio_segment(feature_segment, style_id)
+                )
+                raw_wave = raw_wave_with_margin[
+                    self._core.margin_width * 256 : -self._core.margin_width * 256
+                ]
+                wave = raw_wave_to_output_wave(query, raw_wave, sr_raw_wave)
+                yield wave
+
+        return len(audio_feature) * 256, wave_generator()
```

Comment:
多分`audio_feature`から前後の`margin_width`を引いた値を返す必要があると思います。
```suggestion
        return (
            len(audio_feature) - 2 * self._core.margin_width
        ) * 256, wave_generator()
```